### DISCUSSION TITLE
Difficulties to add a strategy

### DISCUSSION THREAD

**Discussion by faycalrakza** [markusaksli/TradeBot#45] [Ideas]
Hello,

I created my own indicator in Java. I wanted to use your bot because I like the backtesting functionality. The problem is that I don't find where is the strategy. I thought I would find things like "rsi < 30 buy" etc. Can you do a tutorial to add an indicator ? The strategy is simple, it's just "buy under this price" and "take profit here"

Thanks

**Comment by markusaksli** [comment]
It isn't too complicated. This is where the indicators are added
https://github.com/markusaksli/TradeBot/blob/41bb4db7f79194fcb3e48df3b5a837b04699aa18/src/main/java/trading/Currency.java#L52-L54
Then they get checked here
https://github.com/markusaksli/TradeBot/blob/41bb4db7f79194fcb3e48df3b5a837b04699aa18/src/main/java/trading/Currency.java#L142-L144
And the results are checked here
https://github.com/markusaksli/TradeBot/blob/41bb4db7f79194fcb3e48df3b5a837b04699aa18/src/main/java/trading/Currency.java#L135-L137
All of the indicators are here
https://github.com/markusaksli/TradeBot/tree/master/src/main/java/indicators

This is how it currently works but after the rework the config yaml file will contain the indicators you want to use.
